lib/extensions/string_extension.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:ui';
+
+import 'package:collection/collection.dart';
+import 'package:faiadashu/fhir_types/fhir_types.dart';
+import 'package:fhir/primitive_types/fhir_uri.dart';
+import 'package:fhir/r4/basic_types/fhir_extension.dart';
+
+extension StringExtension on String {
+  String translate(List<FhirExtension>? extensions, Locale? locale) {
+    final translationExtension = extensions?.firstWhereOrNull(
+      (transExt) =>
+          transExt.url ==
+              FhirUri('http://hl7.org/fhir/StructureDefinition/translation') &&
+          transExt.extension_?.firstWhereOrNull(
+                (ext) =>
+                    (ext.url == FhirUri('lang')) &&
+                    (ext.valueCode?.value == locale?.languageCode),
+              ) !=
+              null,
+    );
+    return translationExtension?.extension_
+            ?.extensionOrNull('content')
+            ?.valueString ??
+        this;
+  }
```

Comment:
@hanny-ph Thank you for the changes, I'll try to test this myself later.

Just one comment, I see this piece of code replicates to some extent the logic at: https://github.com/sujrd/faiadashu/blob/develop/lib/fhir_types/src/fhir_types_extensions.dart#L148

Do you think it'd be possible to refactor things a bit, so that we can have a single implementation that can be reused in both places?

Reply:
> Do you think it'd be possible to refactor things a bit, so that we can have a single implementation that can be reused in both places?

Yeah, totally agree. Gonna update the implementation.